src/libraries/Microsoft.Extensions.Caching.Abstractions/src/MemoryCacheExtensions.cs

```diff
@@ -178,6 +178,33 @@ public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value,
             return (TItem?)result;
         }
 
+        /// <summary>
+        /// Gets the value associated with this key if it exists, or generates a new entry using the provided key and a value from the given factory if the key is not found.
+        /// </summary>
+        /// <typeparam name="TItem">The type of the object to get.</typeparam>
+        /// <param name="cache">The <see cref="IMemoryCache"/> instance this method extends.</param>
+        /// <param name="key">The key of the entry to look for or create.</param>
+        /// <param name="factory">The factory that creates the value associated with this key if the key does not exist in the cache.</param>
+        /// <param name="createOptions">The <see cref="MemoryCacheEntryOptions"/> needs to be applied to the created <see cref="ICacheEntry"/> if not null.</param>
+        /// <returns>The value associated with this key.</returns>
+        public static TItem? GetOrCreate<TItem>(this IMemoryCache cache, object key, Func<ICacheEntry, TItem> factory, MemoryCacheEntryOptions? createOptions)
```

Comment:
Can we now refactor the existing `GetOrCreate[Async]` to call these new methods?

Reply:
Sure, make sense, thanks very much, updated